src/devices/RGBDRosConversionUtils/rosPixelCode.cpp

```diff
@@ -0,0 +1,88 @@
+/*
+ * Copyright (C) 2006-2020 Istituto Italiano di Tecnologia (IIT)
+ * All rights reserved.
+ *
+ * This software may be modified and distributed under the terms of the
+ * BSD-3-Clause license. See the accompanying LICENSE file for details.
+ */
+
+#include "rosPixelCode.h"
+
+namespace yarp {
+    namespace dev {
+        namespace ROSPixelCode {
+
+std::string yarp2RosPixelCode(int code)
+{
+    switch (code)
+    {
+    case VOCAB_PIXEL_BGR:
+        return BGR8;
+    case VOCAB_PIXEL_BGRA:
+        return BGRA8;
+    case VOCAB_PIXEL_ENCODING_BAYER_BGGR16:
+        return BAYER_BGGR16;
+    case VOCAB_PIXEL_ENCODING_BAYER_BGGR8:
+        return BAYER_BGGR8;
+    case VOCAB_PIXEL_ENCODING_BAYER_GBRG16:
+        return BAYER_GBRG16;
+    case VOCAB_PIXEL_ENCODING_BAYER_GBRG8:
+        return BAYER_GBRG8;
+    case VOCAB_PIXEL_ENCODING_BAYER_GRBG16:
+        return BAYER_GRBG16;
+    case VOCAB_PIXEL_ENCODING_BAYER_GRBG8:
+        return BAYER_GRBG8;
+    case VOCAB_PIXEL_ENCODING_BAYER_RGGB16:
+        return BAYER_RGGB16;
+    case VOCAB_PIXEL_ENCODING_BAYER_RGGB8:
+        return BAYER_RGGB8;
+    case VOCAB_PIXEL_MONO:
+        return MONO8;
+    case VOCAB_PIXEL_MONO16:
+        return MONO16;
+    case VOCAB_PIXEL_RGB:
+        return RGB8;
+    case VOCAB_PIXEL_RGBA:
+        return RGBA8;
+    case VOCAB_PIXEL_MONO_FLOAT:
+        return TYPE_32FC1;
+    default:
+        return RGB8;
+    }
+}
+
+int Ros2YarpPixelCode(std::string roscode)
```

Comment:
Codacy found an issue: [Function parameter 'roscode' should be passed by const reference.](https://app.codacy.com/manual/robotology/yarp/pullRequest?prid=6096993)